### DISCUSSION TITLE
.pre-commit-hooks support

### DISCUSSION THREAD

**Discussion by floatingpurr** [pdm-project/pdm#1247] [Ideas]
Hi folks, I've just started using pdm. It looks great!!! I'm wondering if it's already scheduled an integration with `pre-commit` for exposing pre-cooked hooks.

e.g., a `.pre-commit-hooks.yaml` file with pieces of code like the following:

```
- id: pdm-export
  name: pdm-export-requirements
  description: run pdm export to get requirements.txt
  entry: pdm export
  language: python
  language_version: python3
  pass_filenames: false
  files: ^pdm.lock$
  args: ["-o", "requirements.txt"]
```

**Comment by floatingpurr** [comment]
See #1279

**Comment by gjoseph92** [comment]
Would also be great to have a hook verifying that the lockfile is up to date with `pyproject.toml`. This is highly-requested for poetry, but still hasn't been released: https://github.com/python-poetry/poetry/pull/1954. AFAICT there isn't an equivalent command for PDM though?

**Reply by frostming** [reply]
The closest one is `pdm install --check`, except it performs installation if there is no problem.

**Reply by gjoseph92** [reply]
Seems like `pdm install --check --dry-run` would work here?

**Reply by frostming** [reply]
@gjoseph92 oh yes, I forgot `install` has a `dry-run` option
